src/main/kotlin/app/revanced/extensions/Extensions.kt

```diff
@@ -27,6 +29,30 @@ internal fun MutableMethodImplementation.injectHideCall(
     )
 }
 
+/**
+ * traverse the class hierarchy starting from the given root class
+ *
+ * @param targetClass the class to start traversing the class hierarchy from
+ * @param callback function that is called for every class in the hierarchy
+ */
+fun BytecodeData.traverseClassHierarchy(targetClass: MutableClass, callback: MutableClass.() -> Unit) {
+    callback(targetClass)
+    this.findClass(targetClass.superclass ?: "")?.resolve()?.let {
```

Comment:
Finding a class named `""` is going to get you nowhere.
```suggestion
    this.findClass(targetClass.superclass ?: return)?.resolve()?.let {
```